### ISSUE TITLE
Apply highlights only to the current file

### ISSUE THREAD

**Issue by Otiel** [ryu1kn/vscode-text-marker#11]
Could we have an option to apply highlights to the current file only ?

**Comment by ryu1kn** [comment]
Hi @Otiel , thanks for the suggestion. I'm curious to know why you want this option: i.e. when do you want to highlight patterns only in the current file?

**Comment by Otiel** [comment]
This is how I usually use TextMarker:
1. Open a file (log...) inside vscode without opening a workspace or a folder,
2. Apply highlights to the terms I want to keep in my sight.

But at the same time, I have several other tabs opened in vscode, that have nothing to do with my log file. Thus I don't want to see highlights on those tabs.

Hope that makes sense.

**Comment by ryu1kn** [comment]
I see... but this could be a bit tricky. We need to make whether highlight rules are applied to the current tabs super noticeable; otherwise users may get confused if there are actually no matches or just not enabled.

Maybe for now, you can get around with opening up a separate editor window for the log? It should work until we do #10 ...

**Comment by Otiel** [comment]
Yes indeed. Apart from adding an indicator on the status bar, I'm not sure how you could display the difference.

**Comment by colelawrence** [comment]
I was imagining that you could scope the textmarker highlights to specific file types.
This would be easiest for me, because I could simply add my highlight twice if necessary.

**Comment by ryu1kn** [comment]
@colelawrence That’s an interesting idea 💡 We can introduce a config that allows you to limit the scope of highlight, like `textmarker.limitHighlightToSameFiletype`

**Comment by colelawrence** [comment]
@ryu1kn, yes. Then you might have a command: `TextMarker: Toggle Filetype Limitation`, but in the config:
```json
{
  "textmarker.savedHighlights": [
    {
      "pattern": {
        "type": "string",
        "expression": "Printf",
        "ignoreCase": false,
        "wholeMatch": false,
        "filetypes": [
          "go",
          "typescript",
          "typescriptreact"
        ]
      }
    }
  ]
}
```
